Add GPUCanvasConfiguration.toneMapping.mode

squelart opened the pull request:
#### 60fafaadcbdbefd62d08ae90415f43b3e01ccf89
<pre>
Add GPUCanvasConfiguration.toneMapping.mode
<a href="https://bugs.webkit.org/show_bug.cgi?id=279816">https://bugs.webkit.org/show_bug.cgi?id=279816</a>
<a href="https://rdar.apple.com/136139263">rdar://136139263</a>

Reviewed by Mike Wyrzykowski.

- Add WebGPUHDREnabled internal preference, false by default.
- Implement GPUCanvasToneMapping and GPUCanvasToneMappingMode idl&apos;s (if
  WebGPUHDREnabled), headers and serializations.
- Add GPUCanvasToneMapping to GPUCanvasConfiguration (if WebGPUHDREnabled).
- Transmit tone mapping mode wherever GPUCanvasConfiguration or equivalent
  is used.

If WebGPUHDREnabled is false, the tone mapping mode will just be stuck at
&quot;standard&quot;, so future changes that start using it will not produce any HDR
content.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Headers.cmake:
* Source/WebCore/Modules/WebGPU/GPUCanvasConfiguration.h:
(WebCore::GPUCanvasConfiguration::convertToBacking const):
* Source/WebCore/Modules/WebGPU/GPUCanvasConfiguration.idl:
* Source/WebCore/Modules/WebGPU/GPUCanvasToneMapping.h: Copied from Source/WebCore/Modules/WebGPU/GPUCanvasConfiguration.idl.
* Source/WebCore/Modules/WebGPU/GPUCanvasToneMapping.idl: Copied from Source/WebCore/Modules/WebGPU/GPUCanvasConfiguration.idl.
* Source/WebCore/Modules/WebGPU/GPUCanvasToneMappingMode.h: Copied from Source/WebCore/Modules/WebGPU/GPUCanvasConfiguration.idl.
(WebCore::convertToBacking):
* Source/WebCore/Modules/WebGPU/GPUCanvasToneMappingMode.idl: Copied from Source/WebCore/Modules/WebGPU/GPUCanvasConfiguration.idl.
* Source/WebCore/Modules/WebGPU/Implementation/WebGPUPresentationContextImpl.cpp:
(WebCore::WebGPU::convertToToneMappingMode):
(WebCore::WebGPU::PresentationContextImpl::configure):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPU.serialization.in:
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUCanvasConfiguration.h:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/html/canvas/GPUCanvasContextCocoa.h:
* Source/WebCore/html/canvas/GPUCanvasContextCocoa.mm:
(WebCore::GPUCanvasContextCocoa::reshape):
(WebCore::GPUCanvasContextCocoa::configure):
* Source/WebGPU/WebGPU/PresentationContextIOSurface.h:
* Source/WebGPU/WebGPU/PresentationContextIOSurface.mm:
(WebGPU::PresentationContextIOSurface::configure):
* Source/WebGPU/WebGPU/WebGPU.h:
* Source/WebKit/Scripts/webkit/messages.py:
(headers_for_type):
* Source/WebKit/Shared/WebGPU/WebGPUCanvasConfiguration.cpp:
(WebKit::WebGPU::ConvertToBackingContext::convertToBacking):
(WebKit::WebGPU::ConvertFromBackingContext::convertFromBacking):
* Source/WebKit/Shared/WebGPU/WebGPUCanvasConfiguration.h:
* Source/WebKit/Shared/WebGPU/WebGPUCanvasConfiguration.serialization.in:

Canonical link: <a href="https://commits.webkit.org/283803@main">https://commits.webkit.org/283803@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c64d3c42f5a1635bdcbed370daf87909daf3a296

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/67343 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/46722 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/19975 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/71389 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/18480 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/69461 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/54520 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/18271 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/53971 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/12392 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/70410 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/42912 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/58256 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/34470 "Found 1 new API test failure: /WPE/TestResources:/webkit/WebKitWebView/sync-request-on-max-conns (failure)") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/66855 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/39584 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/15656 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/16838 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/60458 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/61551 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/15997 "Build is in progress. Recent messages:OS: Ventura (13.6.7), Xcode: 14.3; Running apply-patch; Checked out pull request; Skipped layout-tests; 14 flakes") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/73090 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/66588 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/11302 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/15325 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/61445 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/11335 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/58321 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/61501 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14935 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/9256 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/2860 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/88357 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/42527 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/15576 "Found 10 new JSC stress test failures: wasm.yaml/wasm/fuzz/memory.js.wasm-bbq, wasm.yaml/wasm/stress/cc-i32-kitchen-sink.js.wasm-eager, wasm.yaml/wasm/stress/js-to-wasm-many-i64.js.wasm-bbq, wasm.yaml/wasm/stress/js-to-wasm-many-i64.js.wasm-eager-jettison, wasm.yaml/wasm/stress/js-to-wasm-many-i64.js.wasm-slow-memory, wasm.yaml/wasm/stress/js-wasm-call-many-return-types-on-stack-no-args.js.wasm-eager, wasm.yaml/wasm/stress/js-wasm-js-varying-arities.js.wasm-no-cjit, wasm.yaml/wasm/stress/js-wasm-js-varying-arities.js.wasm-slow-memory, wasm.yaml/wasm/stress/tail-call-simple.js.default-wasm, wasm.yaml/wasm/stress/tail-call.js.wasm-eager-jettison (failure)") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/43604 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/44790 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/43345 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mwyrzykowski reviewed (approved):
Looks good - thank you!

squelart committed:
Add GPUCanvasConfiguration.toneMapping.mode
https://bugs.webkit.org/show_bug.cgi?id=279816
rdar://136139263

Reviewed by Mike Wyrzykowski.

- Add WebGPUHDREnabled internal preference, false by default.
- Implement GPUCanvasToneMapping and GPUCanvasToneMappingMode idl's (if
  WebGPUHDREnabled), headers and serializations.
- Add GPUCanvasToneMapping to GPUCanvasConfiguration (if WebGPUHDREnabled).
- Transmit tone mapping mode wherever GPUCanvasConfiguration or equivalent
  is used.

If WebGPUHDREnabled is false, the tone mapping mode will just be stuck at
"standard", so future changes that start using it will not produce any HDR
content.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Headers.cmake:
* Source/WebCore/Modules/WebGPU/GPUCanvasConfiguration.h:
(WebCore::GPUCanvasConfiguration::convertToBacking const):
* Source/WebCore/Modules/WebGPU/GPUCanvasConfiguration.idl:
* Source/WebCore/Modules/WebGPU/GPUCanvasToneMapping.h: Copied from Source/WebCore/Modules/WebGPU/GPUCanvasConfiguration.idl.
* Source/WebCore/Modules/WebGPU/GPUCanvasToneMapping.idl: Copied from Source/WebCore/Modules/WebGPU/GPUCanvasConfiguration.idl.
* Source/WebCore/Modules/WebGPU/GPUCanvasToneMappingMode.h: Copied from Source/WebCore/Modules/WebGPU/GPUCanvasConfiguration.idl.
(WebCore::convertToBacking):
* Source/WebCore/Modules/WebGPU/GPUCanvasToneMappingMode.idl: Copied from Source/WebCore/Modules/WebGPU/GPUCanvasConfiguration.idl.
* Source/WebCore/Modules/WebGPU/Implementation/WebGPUPresentationContextImpl.cpp:
(WebCore::WebGPU::convertToToneMappingMode):
(WebCore::WebGPU::PresentationContextImpl::configure):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPU.serialization.in:
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUCanvasConfiguration.h:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/html/canvas/GPUCanvasContextCocoa.h:
* Source/WebCore/html/canvas/GPUCanvasContextCocoa.mm:
(WebCore::GPUCanvasContextCocoa::reshape):
(WebCore::GPUCanvasContextCocoa::configure):
* Source/WebGPU/WebGPU/PresentationContextIOSurface.h:
* Source/WebGPU/WebGPU/PresentationContextIOSurface.mm:
(WebGPU::PresentationContextIOSurface::configure):
* Source/WebGPU/WebGPU/WebGPU.h:
* Source/WebKit/Scripts/webkit/messages.py:
(headers_for_type):
* Source/WebKit/Shared/WebGPU/WebGPUCanvasConfiguration.cpp:
(WebKit::WebGPU::ConvertToBackingContext::convertToBacking):
(WebKit::WebGPU::ConvertFromBackingContext::convertFromBacking):
* Source/WebKit/Shared/WebGPU/WebGPUCanvasConfiguration.h:
* Source/WebKit/Shared/WebGPU/WebGPUCanvasConfiguration.serialization.in:

Canonical link: https://commits.webkit.org/283803@main
